src/utils/objectUtils.ts

```diff
@@ -94,6 +96,30 @@ export function groupBy<
   )
 }
 
+/**
+ * @param array The array of objects to be grouped.
+ * @param selector The key used for grouping the objects. Support nested keys.
+ * @returns An object where the keys are unique values from the given selector and the values are the corresponding objects from the array.
+ */
+export function groupByPath<T extends object>(array: T[], selector: string): Record<string, T[]> {
+  return array.reduce(
+    (acc, item) => {
+      const key = dotProp.get(item, selector)
+      if (key === undefined || key === null) {
+        return acc
+      }
+      const strKeyPath = key as string
```

Comment:
I wouldn't introduce a new variable just to coerce type, you can do coercing on `const key: string` already if you want